### ISSUE TITLE
Allow custom SqlDriver to support multi-platform encryption

### ISSUE THREAD

**Issue by izakvdhoven** [apollographql/apollo-kotlin#4504]
### Use case
I need to be able to encrypt the normalised cache on non-Android targets (specifically iOS). 

SqlDelight supports configuring a `SqlDriver` for encryption:

``` Kotlin
val configuration = DatabaseConfiguration(
    encryptionConfig = DatabaseConfiguration.Encryption("MyCacheEncryptionKey"),
    ...
)

val driver: SqlDriver = NativeSqliteDriver(configuration)
```

### Describe the solution you'd like
I would like for the `SqlNormalizedCacheFactory` constructor to be made public, to allow supplying a customised `SqlDriver` when instantiating `SqlNormalizedCacheFactory`.

I understand that the constructor is `internal` to allow SqlDelight to remain an implementation detail, but this hides functionality already supported by the dependency. 

Perhaps an overloaded constructor that allows passing an api-agnostic encryption config (eg, `SqlNormalisedCacheEncryptionConfig`) would solve the problem without leaking the SqlDelight dependency?

**Comment by izakvdhoven** [comment]
> the code that creates the drivers is platform specific anyway so there's no need for a `expect constructor(driver: SqlDriver)`, right?

Correct. 

`SqlNormalizedCacheFactory` already has a default constructor (not expect-actual) defined in `androidMain`, `appleMain` and `jvmMain`
```Kotlin
actual class SqlNormalizedCacheFactory internal constructor(
    private val driver: SqlDriver,
) : NormalizedCacheFactory() {
```
Simply making this default constructor public should solve my problem without incurring additional overhead elsewhere.

**Comment by martinbonnin** [comment]
Want to open a PR about this? (you'll most likely have to run `./gradlew apiDump` to update the public API.

**Comment by izakvdhoven** [comment]
Happy to take a stab at it. 

Thanks!

**Comment by martinbonnin** [comment]
Fixed with #4806
